src/models/propulsion/FGPropeller.cpp

```diff
@@ -163,6 +163,12 @@ FGPropeller::FGPropeller(FGFDMExec* exec, Element* prop_element, int num)
   property_name = base_property_name + "/prop-induced-velocity_fps";
   PropertyManager->Tie( property_name.c_str(), this, &FGPropeller::GetInducedVelocity,
                       &FGPropeller::SetInducedVelocity );
+  property_name = base_property_name + "/propeller-power";
+  PropertyManager->Tie( property_name.c_str(), &PowerRequired );
+  property_name = base_property_name + "/propeller-torque";
```

Comment:
Could you please append the unit name to the property names ? We are trying to display the units in which the quantity is given so that users can implement conversion factors when needed. For example the propeller induced velocity on line 163 is `prop-induced-velocity_fps` where `_fps` means `ft/s`.

The unit of Power is `slug*ft^2*s^-3` in `FGPropeller`
https://github.com/JSBSim-Team/jsbsim/blob/77ecd97fa759ab0b8d1fbb33a6ea1748d3aa6732/src/models/propulsion/FGPropeller.cpp#L358
and the unit of Torque is `slug*ft^2*s^-2.`
https://github.com/JSBSim-Team/jsbsim/blob/77ecd97fa759ab0b8d1fbb33a6ea1748d3aa6732/src/models/propulsion/FGPropeller.cpp#L272
I am not much familiar with Imperial/British/US units for power and torques so there might be shorter names for the units above and you are welcome to use them :smiley:

Reply:
I agree about the Imperial units, its a dreadful state. Should I only add units to the scope of this PR (the 3 new property tree elements), or throughout more of the code?

The variable naming and units is actually why I didn't want to touch FGElectric with this PR, there is a variable 'HP' and 'PowerWatt'; HP is the current power (in horsepower of course) and PowerWatt, is the maximum power.

Reply:
> Should I only add units to the scope of this PR (the 3 new property tree elements), or throughout more of the code?

Oops, you just need to update the 3 new properties you are adding. Unfortunately regarding the other properties it's already too late because renaming would break backward compatibility.
> The variable naming and units is actually why I didn't want to touch FGElectric with this PR, there is a variable 'HP' and 'PowerWatt'; HP is the current power (in horsepower of course) and PowerWatt, is the maximum power.

Indeed, we have not always been consistent about internal usage of SI units vs Imperial units. I guess people tend to use the system unit they feel more comfortable with and convert units from/to the Imperial system when needed.

Reply:
Another option is to add methods for conversion to another unit (for example `FGPropeller::GetPowerRequiredHP()` or `FGPropeller::GetPowerRequiredWatt()`) and use these methods to output the power in the corresponding unit.
You can even create a different property for each unit:
```c++
  property_name = base_property_name + "/propeller-power_watt";
  PropertyManager->Tie( property_name.c_str(), &PowerRequiredWatt );
  property_name = base_property_name + "/propeller-power_hp";
  PropertyManager->Tie( property_name.c_str(), &PowerRequiredHP );
```
Do as you see fit your need as long as the unit name is specified in the property name.